draft-ietf-quic-http.md

```diff
@@ -201,6 +201,21 @@ additional framing. Note that a request or response without a body will cause
 this stream to be half-closed in the corresponding direction without
 transferring data.
 
+Because the message control stream contains HPACK data which manipulates
+connection-level state, the message control stream MUST NOT be closed with a
+stream-level error.  If an implementation chooses to reject a request with a
+QUIC error code, it MUST trigger a RST_STREAM on the data stream only.  An
+implementation MAY close (FIN) a message control stream without completing a
+full HTTP message if the data stream has been abruptly closed.  Data on message
+control streams MUST be fully consumed, or the connection terminated.
+
+If a message control stream is terminated abruptly by the transport layer (e.g.
+with QUIC_TOO_MANY_OPEN_STREAMS), any frames already sent on that stream MUST
+be re-played on another message control stream as soon as the maximum number of
+streams permits.  If the implementation no longer requires the response to this
+request, it MAY abruptly terminate the corresponding data stream at the same
+time.
```

Comment:
Anyone who exceeds a stream limit is violating the protocol (we need to address the issue of squishiness in that limit).  And refusing a stream is easy: refuse the data stream.  If the concern is that headers will consume stream limit until they resolve, I think that's a feature, not a bug.

0-RTT streams that end up exceeding some server limit can be refused by hitting the data stream.  The control stream can be left to run to completion.  If the server can't handle that (because the limits have been reduced by more than 50% perhaps), then it probably should reject 0-RTT.

A server might have to reject the connection (after all, it might only discover this sort of dire condition after foolishly/optimistically accepting 0-RTT), but that means that a new connection will have to created by the client.  Mike already has text about needing to retry in #236.  In the end, I think that sort of retry is something clients that attempt 0-RTT will just have to deal with.